File: LeetCode/283.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        vector<int>::size_type i = 0;
        while (i < nums.size()){
            if (nums[i] == 0){
                vector<int>::size_type j = i + 1;
                while ((j < nums.size()) and (nums[j] == 0)) j++;
                if (j != nums.size()){
                    int tmp = nums[i];
                    nums[i] = nums[j];
                    nums[j] = tmp;
                }
            }
            i += 1;
        }
    }
};

File: LeetCode/283.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = 0
        while i < len(nums):
            if not(nums[i]):
                j = i + 1
                while j < len(nums) and not(nums[j]):
                    j += 1
                if j != len(nums):
                    nums[i], nums[j] = nums[j], nums[i]
            i += 1
        return